File: ViewControllerDemo/ViewControllerDemo/ViewController.m
//
//  ViewController.m
//  ViewControllerDemo
//
//  Created by pfl on 16/1/6.
//  Copyright © 2016年 pfl. All rights reserved.
//

#import "ViewController.h"
#import "PushViewController.h"
#import "PresentViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIButton *btn = [[UIButton alloc]initWithFrame:CGRectMake(0, 0, 200, 44)];
    btn.center = CGPointMake(CGRectGetMidX(self.view.bounds), CGRectGetMidY(self.view.bounds)-20);
    [btn setTitle:@"push" forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(btnPressed:) forControlEvents:UIControlEventTouchUpInside];
    btn.tag = 1;
    [btn setBackgroundColor:[UIColor orangeColor]];
    
    
    UIButton *btn2 = [[UIButton alloc]initWithFrame:CGRectMake(0, 0, 200, 44)];
    btn2.center = CGPointMake(CGRectGetMidX(self.view.bounds), CGRectGetMidY(self.view.bounds)+20);
    [btn2 setTitle:@"present" forState:UIControlStateNormal];
    [btn2 addTarget:self action:@selector(btnPressed:) forControlEvents:UIControlEventTouchUpInside];
    btn2.tag = 2;
    [btn2 setBackgroundColor:[UIColor redColor]];
    
    [self.view addSubview:btn];
    [self.view addSubview:btn2];

    
}

- (void)btnPressed:(UIButton*)btn {
    if (btn.tag == 1) {
        PushViewController *push = [[PushViewController alloc]initWithNibName:nil bundle:nil];
        [self.navigationController pushViewController:push animated:YES];
    }
    else {
        PresentViewController *present = [[PresentViewController alloc]init];
        
        
        UINavigationController *nav = [[UINavigationController alloc]initWithRootViewController:present];
        
        [self presentViewController:nav animated:YES completion:nil];
 
        NSLog(@"====presentedViewControlle1:%@",nav);
        NSLog(@"====presentedViewControlle2:%@",self.presentedViewController);
        NSLog(@"====presentedViewControlle2-1:%@",present.presentingViewController);

        NSLog(@"====presentingViewController3:%@",nav.presentingViewController);
        NSLog(@"====presentingViewController4:%@",self.navigationController);


    }
}


@end

File: CustomNavigationBarDemo/CustomNavigationBarDemo/ViewController.swift
//
//  ViewController.swift
//  CustomNavigationBarDemo
//
//  Created by pfl on 16/2/5.
//  Copyright © 2016年 pfl. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var customNavigationBar: UINavigationBar!
    private var customNavigationItem: UINavigationItem = UINavigationItem(title: "Detail")
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        customNavigationBar = UINavigationBar(frame: CGRectMake(0,0,UIScreen.mainScreen().bounds.width,64))
        customNavigationBar.setItems([customNavigationItem], animated: false)
        customNavigationBar.tintColor = UIColor.whiteColor()
        customNavigationBar.translucent = true
        customNavigationBar.barStyle = .BlackTranslucent
        customNavigationBar.shadowImage = UIImage()
        customNavigationBar.backgroundColor = UIColor.clearColor()
        customNavigationBar.setBackgroundImage(UIImage(), forBarMetrics: .Default)
        customNavigationBar.tintAdjustmentMode = .Normal
        view.addSubview(customNavigationBar)
        
        let titleTextAttributes = [NSForegroundColorAttributeName: UIColor.whiteColor()]
        customNavigationBar.titleTextAttributes = titleTextAttributes
        
        
        
        let btn = UIButton(frame: CGRect(x: 0, y: 0, width: 200, height: 50))
        btn.center.y = 200
        btn.center.x = self.view.center.x
        btn.addTarget(self, action: Selector("push"), forControlEvents: .TouchUpInside)
        view.addSubview(btn)
        btn.setTitle("push", forState: .Normal)
        btn.setTitleColor(UIColor.orangeColor(), forState: .Normal)
        
        
        let settingsBarButtonItem = UIBarButtonItem(image: UIImage(named: "icon_settings"), style: .Plain, target: self, action: "showSettings:")
        
        customNavigationItem.rightBarButtonItem = settingsBarButtonItem

        
    }
    
    func showSettings(item:UINavigationItem) {
        navigationController?.pushViewController(NextViewController(), animated: true)
    }
    
    
    func push() {
//        navigationController?.pushViewController(NextViewController(), animated: true)
    }

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
//        navigationController?.navigationBarHidden = true
    }

    
    
    

}

